sycl/source/detail/pi.cpp

```diff
@@ -61,10 +99,19 @@ void initialize() {
   if (!useBackend(SYCL_BE_PI_OPENCL)) {
     die("Unknown SYCL_BE");
   }
-  #define _PI_API(api)                          \
-    extern decltype(::api) * api##OclPtr; \
-    api = api##OclPtr;
-  #include <CL/sycl/detail/pi.def>
+
+  std::string PluginPath;
+  if (!findPlugin(PluginPath))
+    die("Plugin Not Found.");
+
+  void *Library = loadPlugin(PluginPath);
+  if (!Library) {
+    die("Failed to load plugin. Check if plugin is present.");
```

Comment:
Please add the name(s)/path(s) of the plugins not found here and at the bind-call to the error message to help figuring out what is going on.

Reply:
Done.
Could not list the PI API specifically that could not be found in the library. 
The binding logic will be changed in the next patch. I will try to take care of better debugging messages in that case.

Reply:
Awesome, thank you! That will save quite a bit of time hunting down misplaced plugins in the future :smiley: